napari/_qt/_qapp_model/_tests/test_layerlist_context_actions.py

```diff
@@ -0,0 +1,29 @@
+import pytest
+
+from napari._app_model import get_app
+from napari._app_model.actions._layerlist_context_actions import (
+    LAYERLIST_CONTEXT_ACTIONS,
+)
+from napari.components.layerlist import LayerList
+
+
+@pytest.mark.parametrize('layer_action', LAYERLIST_CONTEXT_ACTIONS)
+def test_layer_actions_ctx_menu_execute_command(layer_action):
+    app = get_app()
+    command_id = layer_action.id
+
+    def provide_layer_list() -> 'LayerList':
+        return LayerList()
+
+    with app.injection_store.register(providers=[(provide_layer_list,)]):
```

Comment:
So I don't think you need to do this, the layer list provider:

https://github.com/napari/napari/blob/f0c02a844cdd0eebceb64b696bb4ab73c2aed08f/napari/_qt/_qapp_model/injection/_qproviders.py#L95-L96

uses the `_provide_viewer` provider. And I think using `make_napari_viewer` will give you a viewer.

If you did want the test to specifically use your provider, you can add a `weight` arg to `providers` param.

See this comment thread: https://github.com/napari/napari/pull/4991#discussion_r1513648370

I think in this case we want the test to use the `_provide_active_layer_list` provider.

Reply:
But otherwise this looks good to me!

Also just to confirm, all the layer action functions are already tested over with the headless files right?

Reply:
> all the layer action functions are already tested over with the headless files right?

After I quick check I'm not totally sure 😅 I think the functions being tested are the ones imported at:

https://github.com/napari/napari/blob/f0c02a844cdd0eebceb64b696bb4ab73c2aed08f/napari/layers/_tests/test_layer_actions.py#L8-L18

So seems like there are a couple of functions without tests like `_split_stack`, `_split_rgb` and `_merge_stack`? 🤔

Reply:
Agree with @DragaDoncila that these function tests can be added later.

I forgot to say that you can add a layer to the viewer and hopefully this will prevent the errors. You can see [this test](https://github.com/napari/napari/blob/f0c02a844cdd0eebceb64b696bb4ab73c2aed08f/napari/_qt/_qapp_model/_tests/test_file_menu.py#L295) for an example. Ideally we'd get the commands to run without error, so we make sure the provider works

Reply:
> Ideally we'd get the commands to run without error, so we make sure the provider works

I think these tests still make sure the provider works, as the errors would only be raised if the function got to run right? But I don't mind either way if we have layers in there or not

Reply:
Oh yeah you are right. And looking at the tests that raise errors, you'd have to add >1 layer to get them to work so this is good as is. Ignore my comment!